linked-list.rb

```diff
@@ -63,14 +63,45 @@ def display
    end
 
    def include?(key)
+     current = @head
+    #while we have nodes to traverse
+     while current != nil
+     #when we find the key
+       return true if current.value == key
+       #reassign currrent
+       current = current.next_node
+     end
+     puts "The list does not include #{key}"
+     return false
    end
 
    def size
+    @size = 0
+    current = @head
+     while current != nil
+       #increase size
+       @size += 1
+       #reassign current
+       current = current.next_node
+     end
+    puts "This linked list has #{@size} elements."
+    return @size
    end
 
    def max
+     current = @head
+     max_value = current.value
+      while current != nil
+        #if the current node has a larger value than the max gets reasigned
+        if current.value > max_value
+          max_value = current.value
+        end
+        #if max_value is larger than keep moving to check next node
+        current = current.next_node
+      end
+      puts "#{max_value} is the max value element in the list."
+      return max_value
    end
```

Comment:
This looks good! Though I'm not sure what Ruby does in the case that you call `.value` on a `nil` object if there are no nodes in the list at all (@head == nil). In other languages this causes an error because you can't call a method on nil. Therefore I don't know what max_value is in the case thta the list is empty. Regardless, you should no that this is something we should protect against happening in other languages.